apps/frontend/src/app/5_pages/BitocracyPage/components/NewProposalForm/components/ParametersStep/ParametersStep.utils.ts

```diff
@@ -0,0 +1,99 @@
+import { ethers } from 'ethers';
+
+import { ProposalCreationParameter } from '../../../../contexts/ProposalContext.types';
+import { ProposalContract } from '../../NewProposalForm.types';
+import {
+  LOAN_TOKEN_LOGIC_LM_OPTIONS,
+  LOAN_TOKEN_LOGIC_WRBTC_OPTIONS,
+  LOAN_TOKEN_OPTIONS,
+  SOVRYN_PROTOCOL_OPTIONS,
+  STAKING_CONTRACT_OPTIONS,
+} from './ParametersStep.constants';
+import { ContractDetailsResponse } from './ParametersStep.types';
+
+export const getParameterOptions = (contractName: string) => {
+  switch (contractName) {
+    case ProposalContract.SovrynProtocol:
+      return SOVRYN_PROTOCOL_OPTIONS;
+    case ProposalContract.Staking:
+      return STAKING_CONTRACT_OPTIONS;
+    case ProposalContract.LoanTokenLogicLM:
+      return LOAN_TOKEN_LOGIC_LM_OPTIONS;
+    case ProposalContract.LoanTokenLogicWRBTC:
+      return LOAN_TOKEN_LOGIC_WRBTC_OPTIONS;
+    default:
+      return LOAN_TOKEN_OPTIONS;
+  }
+};
+
+export const getContractDetails = (
+  contractName: string,
+): ContractDetailsResponse => {
+  switch (contractName) {
+    case ProposalContract.SovrynProtocol:
+      return { contractName: 'protocol', contractGroup: 'protocol' };
+    case ProposalContract.Staking:
+      return { contractName: 'staking', contractGroup: 'protocol' };
+    case ProposalContract.LoanTokenLogicLM:
+      return {
+        contractName: 'loanTokenLogicBeaconLM',
+        contractGroup: 'protocol',
+      };
+    case ProposalContract.LoanTokenLogicWRBTC:
+      return {
+        contractName: 'loanTokenLogicBeaconWrbtc',
+        contractGroup: 'protocol',
+      };
+    case ProposalContract.iDOC:
+      return { contractName: 'doc', contractGroup: 'loanTokens' };
+    case ProposalContract.iUSDT:
+      return { contractName: 'usdt', contractGroup: 'loanTokens' };
+    case ProposalContract.iRBTC:
+      return { contractName: 'rbtc', contractGroup: 'loanTokens' };
+    case ProposalContract.iBPRO:
+      return { contractName: 'bpro', contractGroup: 'loanTokens' };
+    default:
+      return { contractName: 'xusd', contractGroup: 'loanTokens' };
+  }
+};
+
+function encodeParameters(types: string[], values: any[]): string {
+  const coder = new ethers.utils.AbiCoder();
+  return coder.encode(types, values);
+}
+
+export const renderCalldata = (
+  contract: string,
+  value: any,
+  type: string[],
+  isToken: boolean,
+  token: string,
+  parameterName: string,
+) => {
+  try {
+    const parameters =
+      isToken && parameterName === 'checkPause'
+        ? [contract, token, value]
+        : [contract, value];
+    const encodedValue = encodeParameters(['address', ...type], parameters);
```

Comment:
You don't need to pass extra params to encode it. Pass just types/values that are required by function, no additional ones.

```suggestion
    const parameters =
      isToken && parameterName === 'checkPause'
        ? [token, value]
        : [value];
    const encodedValue = encodeParameters(type, parameters);
```

Also,`checkPause` aka `toggleFunctionPause(string, bool)` want's to have string and bool, you are passing `token` which address and `value` which is bool. 

To implement it correctly you should actually have 2 input fields for 2 values here (one is string and one is boolean), because it expects to have function name and not token address you want to pause.

Also, user is likely to make a mistake here as it's not clear what kind of values he can use here.